src/sonic-config-engine/sonic_device_util.py

```diff
@@ -64,7 +84,11 @@ def get_system_mac():
         last_byte = mac[-2:]
         aligned_last_byte = format(int(int(last_byte, 16) & 0b11000000), '02x')
         mac = mac[:-2] + aligned_last_byte
-    return mac
+
+    if validate_mac_address(mac):
```

Comment:
>validate_mac_address [](start = 7, length = 20)

Better to check valid immediately after `mac = mac.strip()`. It will protect more code.